src/engine/Server.cpp

```diff
@@ -469,6 +469,23 @@ nlohmann::json Server::composeCacheStatsJson() const {
   return result;
 }
 
+// _____________________________________________
+
+ad_utility::websocket::common::OwningQueryId getQueryId(
+    const ad_utility::httpUtils::HttpRequest auto& request) {
+  using ad_utility::websocket::common::OwningQueryId;
+  std::string_view queryIdHeader = request.base()["Query-Id"];
+  if (queryIdHeader.empty()) {
+    return OwningQueryId::uniqueId();
+  }
+  auto queryId = OwningQueryId::uniqueIdFromString(std::string(queryIdHeader));
+  if (!queryId) {
+    throw std::runtime_error{"QueryId '" + queryIdHeader +
```

Comment:
This currently results in a 400 Bad Request reponse. Not sure if it's appropriate in this case

Reply:
I think 409 would be appropriate here. According to [RFC9110](https://www.rfc-editor.org/rfc/rfc9110#status.409):
> The 409 (Conflict) status code indicates that the request could not be completed due to a conflict with the current state of the target resource. This code is used in situations where the user might be able to resolve the conflict and resubmit the request. The server SHOULD generate content that includes enough information for a user to recognize the source of the conflict.
>
> Conflicts are most likely to occur in response to a PUT request. For example, if versioning were being used and the representation being PUT included changes to a resource that conflict with those made by an earlier (third-party) request, the origin server might use a 409 response to indicate that it can't complete the request. In this case, the response representation would likely contain information useful for merging the differences based on the revision history.

400 status codes usually indicate syntactic or logical errors with the request per [RFC9110](https://httpwg.org/specs/rfc9110.html#status.400):
> The 400 (Bad Request) status code indicates that the server cannot or will not process the request due to something that is perceived to be a client error (e.g., malformed request syntax, invalid request message framing, or deceptive request routing).